config/mariadb/config.go

```diff
@@ -0,0 +1,124 @@
+				xpv1.ResourceCredentialsSecretEndpointKey: []byte(attr["fqdn"].(string)),
+				xpv1.ResourceCredentialsSecretPortKey:     []byte(strconv.Itoa(mariadbServerPort)),
+			}, nil
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_database", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/databases/database1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB",
+			"servers", "server_name",
+			"databases", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_firewall_rule", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/firewallRules/rule1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB",
+			"servers", "server_name",
+			"firewallRules", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_virtual_network_rule", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+			"subnet_id": config.Reference{
+				Type:      rconfig.SubnetReferencePath,
+				Extractor: rconfig.ExtractResourceIDFuncPath,
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myresourcegroup/providers/Microsoft.DBforMariaDB/servers/myserver/virtualNetworkRules/vnetrulename
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.DBforMariaDB",
+			"servers", "server_name",
+			"virtualNetworkRules", "name",
+		)
+	})
+
+	p.AddResourceConfigurator("azurerm_mariadb_configuration", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.References = config.References{
+			"server_name": config.Reference{
+				Type: "Server",
+			},
+		}
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/mygroup1/providers/Microsoft.DBforMariaDB/servers/server1/configurations/interactive_timeout
+		// which needs to be a valid MariaDB configuration name
+		// See https://mariadb.com/kb/en/server-system-variables/
+		r.ExternalName = config.IdentifierFromProvider
```

Comment:
I think `config.NameAsIdentifier` better suits this resource. I checked what we did in some other RDBMS server resources for which we have added a configuration resource. For postgresql, we are using `config.IdentifierFromProvider` like you did here. The reasoning could be the configuration option name may not be valid RFC1123 DNS subdomain name. But then we have the external-name annotation through which the user can specify it. But then one may argue it's not the greatest UX (but then having a `spec.name` is not good for XRM, external-name should be the authoritative source for managed resources).

Reply:
@ulucinar Yes, for me RFC1123 was exactly the blocker for `config.NameAsIdentifier` . If we follow your suggestion t it will mean mandatory external-name annotation specification, is that correct? If it is fine, I can change it